File: app/src/main/java/fobo66/exchangecourcesbelarus/list/BestCoursesAdapter.java
package fobo66.exchangecourcesbelarus.list;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import fobo66.exchangecourcesbelarus.R;
import fobo66.exchangecourcesbelarus.models.BestCourse;
import java.util.ArrayList;
import java.util.List;

/**
 * Adapter for currency cards
 *
 * Created by fobo66 on 23.08.2015.
 */
public class BestCoursesAdapter extends RecyclerView.Adapter<CurrencyViewHolder> {

  private List<BestCourse> courses = new ArrayList<>();
  private boolean buyOrSell;

  public BestCoursesAdapter(List<BestCourse> courses) {
    this.courses.addAll(courses);
  }

  @NonNull @Override
  public CurrencyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
    View v =
        LayoutInflater.from(parent.getContext()).inflate(R.layout.currency_card, parent, false);
    return new CurrencyViewHolder(v);
  }

  @Override public void onBindViewHolder(@NonNull CurrencyViewHolder holder, int position) {
    BestCourse bestCourse = courses.get(position);

    if (bestCourse.isBuy == buyOrSell) {
      holder.bind(bestCourse);
    }
  }

  @Override public int getItemCount() {
    return courses.size();
  }

  public void onDataUpdate(List<BestCourse> newCourses) {
    courses.clear();
    courses.addAll(newCourses);
    notifyDataSetChanged();
  }

  public void setBuyOrSell(boolean buyOrSell) {
    this.buyOrSell = buyOrSell;
  }
}

File: app/src/main/java/fobo66/exchangecourcesbelarus/list/BestCoursesAdapter.kt
package fobo66.exchangecourcesbelarus.list

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView.Adapter
import fobo66.exchangecourcesbelarus.R.layout
import fobo66.exchangecourcesbelarus.entities.BestCourse

/**
 * Adapter for currency cards
 * Created by fobo66 on 23.08.2015.
 */
class BestCoursesAdapter(courses: List<BestCourse>) :
  Adapter<CurrencyViewHolder>() {
  private val courses: MutableList<BestCourse> = ArrayList()
  private var buyOrSell = false
  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CurrencyViewHolder {
    val v =
      LayoutInflater.from(parent.context).inflate(layout.currency_card, parent, false)
    return CurrencyViewHolder(v)
  }

  override fun onBindViewHolder(holder: CurrencyViewHolder, position: Int) {
    val bestCourse = courses[position]
    if (bestCourse.isBuy == buyOrSell) {
      holder.bind(bestCourse)
    }
  }

  override fun getItemCount(): Int {
    return courses.size
  }

  fun onDataUpdate(newCourses: List<BestCourse>) {
    courses.clear()
    courses.addAll(newCourses)
    notifyDataSetChanged()
  }

  fun setBuyOrSell(buyOrSell: Boolean) {
    this.buyOrSell = buyOrSell
  }

  init {
    this.courses.addAll(courses)
  }
}